### DISCUSSION TITLE
[Feature Request] lion/snackbar component

### DISCUSSION THREAD

**Discussion by MathieuPuech** [ing-bank/lion#918] [General]
Like material design snackbars
> Snackbars provide brief messages about app processes at the bottom of the screen.

Rules:
- Only one snackbar may be displayed at a time.
- Disappear automatically
- Not dismissible on outside click or escape
- Should not contain actions (see this [article](https://www.scottohara.me/blog/2019/07/08/a-toast-to-a11y-toasts.html))

A `show` function should be exported to instanciate the component display

**Comment by MathieuPuech** [comment]
Some questions:
- is snackbar the good name?
- do we allow message to be text only or do we allow a template? (if we want icons inside for example)

**Comment by erikkroes** [comment]
The trouble with these components is the timing combined with the content.

Your content can't be so important that people can't miss it. If people can't miss it, you would pick something that doesn't disappear (for example). But if the content is of so little importance that you can put it in a snackbar, couldn't you do without it entirely?
What would your usecase for a snackbar be?

**Comment by MathieuPuech** [comment]
Personally, I use snackbar for situations described in [material design confirmations & acknowledgement](https://material.io/design/communication/confirmation-acknowledgement.html#usage)

> Acknowledging actions provide text to let the user know whether an action they chose has been completed.

It could be, "build complete", "saved", "removed".

And like it was [used in pwa-starter-kit](https://github.com/Polymer/pwa-starter-kit/blob/master/src/components/my-app.js#L235) for offline/online notifications.

Saved and removed could be inline notification (like tooltip next to the action button) but you can validate a form without clicking on the submit button. And build complete is linked to a notification system, it's something done in background.
